src/cljs/rems/paging.cljs

```diff
@@ -0,0 +1,101 @@
+(ns rems.paging
+  (:require [reagent.core :as r]
+            [rems.atoms :as atoms]
+            [rems.guide-util :refer [component-info example]]
+            [rems.text :refer [text]]))
+
+(defn- page-number [paging page on-change]
+  (if (= (:current-page paging) page)
+    [atoms/link {:label [:u (str (inc page))]
+                 :class "btn btn-link current-page"
+                 :disabled true
+                 :on-click #(on-change (assoc paging :current-page page))}]
+
+    [atoms/link {:label (str (inc page))
+                 :class "btn btn-link"
+                 :on-click #(on-change (assoc paging :current-page page))}]))
+
+(defn paging-field
+  "Component for showing page numbers.
+
+  Intended to be used together with a `rems.table/table` through `rems.table/paging`.
+
+  `:id`                      - identity of the component (derived from table)
+  `:on-change`               - callback
+  `:paging`                  - paging state (with table)
+    `:current-page`          - the current page (0-indexed)
+    `:show-all-page-numbers` - state of whether to show all page numbers or `...`
+  `:pages`                   - how many pages exist"
+  [{:keys [id on-change paging pages]}]
+  (r/with-let [show-all-page-numbers (r/atom (:show-all-page-numbers paging))]
+    (when (> pages 1)
+      [:div.d-flex.gap-1.align-items-center.justify-content-center.flex-wrap
+       [:div (text :t.table.paging/page)]
+       [:div.mr-3.my-3
+        {:style (if @show-all-page-numbers
+                  {:display :grid
+                   :grid-template-columns "repeat(20, min-content)"
+                   :align-items :baseline}
+                  {:display :flex
+                   :justify-content :center
+                   :align-items :center
+                   :flex-wrap :wrap})
+         :id (str id "-pages")}
+
+        (if (or @show-all-page-numbers
+                (< pages 10))
+          ;; just show them all
+          (for [page (range pages)]
+            ^{:key (str id "-page-" page)}
+            [page-number paging page on-change])
+
+          ;; show 1 2 3 ... 7 8 9
+          (let [first-pages (take 3 (range pages))
+                last-pages (take-last 3 (drop 3 (range pages)))]
+            [:<>
+             (for [page first-pages]
+               ^{:key (str id "-page-" page)}
+               [page-number paging page on-change])
+
+             ^{:key (str id "-page-...")}
+             [atoms/link {:label "..."
+                          :class "btn btn-link"
+                          :on-click #(reset! show-all-page-numbers true)}]
+
+             (for [page last-pages]
+               ^{:key (str id "-page-" page)}
+               [page-number paging page on-change])]))]])))
```

Comment:
(original comment below, modified related lines for better review context here)

page render loop could be extracted?

Reply:
Maybe it makes sense. I initially thought that it doesn't as the code is so basic that it wouldn't help much.